lib/web/device_trust_test.go

```diff
@@ -164,3 +170,76 @@ func (f *fakeDevicesClient) resetConfirmRequests() []*devicepb.ConfirmDeviceWebA
 
 	return reqs
 }
+
+func TestHandler_GetRedirectURL(t *testing.T) {
+	h := &Handler{}
+
+	tests := []struct {
+		name           string
+		redirectURI    string
+		expectedURL    string
+		expectedErrMsg string
+	}{
+		{
+			name:        "no redirect_uri",
+			redirectURI: "",
+			expectedURL: "/web",
+		},
+		{
+			name:        "with redirect_uri",
+			redirectURI: "https://example.com/web/custom/path",
+			expectedURL: "/web/custom/path",
+		},
+		{
+			name:        "with app access redirect_uri",
+			redirectURI: "https://example.com/web/launch/myapp.example.com",
+			expectedURL: "/web/launch/myapp.example.com",
+		},
+		{
+			name:           "with invalid redirect_uri",
+			redirectURI:    "://invalid",
+			expectedURL:    "/web",
+			expectedErrMsg: "parse \"://invalid\": missing protocol scheme",
+		},
+		{
+			name:        "with external redirect_uri",
+			redirectURI: "https://example.com/path",
+			expectedURL: "/web/path",
+		},
+		{
+			name:        "with empty path redirect_uri",
+			redirectURI: "https://example.com",
+			expectedURL: "/web",
+		},
+		{
+			name:        "with relative path",
+			redirectURI: "/custom/path",
+			expectedURL: "/web/custom/path",
+		},
+		{
+			name:        "with web prefix already",
+			redirectURI: "/web/existing/path",
+			expectedURL: "/web/existing/path",
+		},
+		{
+			name:           "with completely invalid URI",
+			redirectURI:    "not a URI at all",
+			expectedURL:    "/web",
+			expectedErrMsg: "parse \"not a URI at all\": invalid URI for request",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			result, err := h.getRedirectURL(tt.redirectURI)
```

Comment:
Isn't this tested as part of TestHandler_DeviceWebConfirm? If we can test via the public method that typically makes for a better test than testing a private helper directly.

Reply:
Removed this test and added missing cases to `TestHandler_DeviceWebConfirm`